src/daemons/rhsmcertd.c

```diff
@@ -234,34 +236,97 @@ long long gen_random(long long max) {
     return random_num % true_max;
 }
 
-/* Handle program signals */
+/**
+ * Try to run Python script package-profile-uploader. This script tries to upload DNF profile
+ * to server, when server supports profile
+ * @return Return true, when uploading was successful. Otherwise, return false.
+ */
+static gboolean
+upload_package_profile ()
+{
+    int status = 0;
+
+    int pid = fork ();
+    if (pid < 0) {
+        error ("Fork failed");
+        exit (EXIT_FAILURE);
+    }
+    if (pid == 0) {
+        debug ("Uploading of package profile (executing: %s)", PACKAGE_PROFILE_UPLOADER);
+        execl (PACKAGE_PROFILE_UPLOADER, PACKAGE_PROFILE_UPLOADER_NAME, NULL);
+    }
+
+    waitpid (pid, &status, 0);
+    status = WEXITSTATUS (status);
+
+    if (status == 0) {
+        info ("Uploading of package profile performed successfully.");
+        // No need to repeat this action again
+        return true;
+    } else {
+        info ("Uploading of package profile terminated with status code: %d", status);
+        return false;
+    }
+}
+
+
+/**
+ * Handle program signals. This method can handle two signals SIGTERM and SIGUSR1
+ */
 void
 signal_handler(int signo) {
+    debug ("Handling signal %d", signo);
     if (signo == SIGTERM) {
+        info ("rhsmcertd is shutting down...");
         /* Close lock file and release lock on this file */
         if (fd_lock != -1) {
+            /* Truncate lock file to zero value (delete PID before shutting down) */
+            int ret;
+            ret = ftruncate(fd_lock, 0);
+            if (ret == -1) {
+                warn ("Unable to truncate file: %s, %s", LOCKFILE, strerror (errno));
+            }
             close(fd_lock);
             fd_lock = -1;
         }
-        info ("rhsmcertd is shutting down...");
         signal (signo, SIG_DFL);
         raise (signo);
     }
+    if (signo == SIGUSR1) {
+        /* When user defined signal 1 is received, then try to upload package profile */
+        debug ("Received SIGUSR1 signal");
+        upload_package_profile();
+    }
 }
 
+/**
+ * Try to lock file /var/lock/subsys/rhsmcertd and write current PID to this file
+ * @return Return 0, when it was possible lock file and write PID to lock file.
+ *         Otherwise, return non-zero value.
+ */
 int
 get_lock ()
 {
     fd_lock = open (LOCKFILE, O_WRONLY | O_CREAT, 0640);
     int ret = 0;
 
     if (fd_lock == -1) {
+        debug ("Unable to open file: %s", LOCKFILE);
         ret = 1;
     } else {
         if (flock (fd_lock, LOCK_EX | LOCK_NB) == -1) {
+            debug ("Unable to lock file: %s", LOCKFILE);
             close (fd_lock);
             fd_lock = -1;
-            ret = 1;
+            ret = 2;
+        } else {
+            pid_t pid = getpid();
```

Comment:
nitpick: missing space before the opening bracket, as it seems the style of this code